### PR TITLE
Merging sequencing module into genome module 

### PR DESCRIPTION
**Pull request by josemduarte**
This solves #536. 

I did some warning cleanup as well.

### CODE REVIEW

File: biojava-genome/src/main/java/org/biojava/nbio/genome/io/fastq/AbstractFastqReader.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
  *      http://www.biojava.org/
  *
  */
-package org.biojava.nbio.sequencing.io.fastq;
+package org.biojava.nbio.genome.io.fastq;
```
Comment (heuermh) [inline biojava-genome/src/main/java/org/biojava/nbio/genome/io/fastq/AbstractFastqReader.java]: Should all these classes moved to a new package have `@since 5.0.0` tags?  Note there are other `@since` tags present on classes and methods throughout the old `o.b.n.sequencing.io.fastq` package.

File: biojava-genome/src/test/java/org/biojava/nbio/genome/io/fastq/AbstractFastqReaderTest.java [hunk 2]
```diff
@@ -28,6 +28,11 @@
 import java.io.StringReader;
 import java.net.URL;
 
+import org.biojava.nbio.genome.io.fastq.Fastq;
```
Comment (heuermh) [inline biojava-genome/src/test/java/org/biojava/nbio/genome/io/fastq/AbstractFastqReaderTest.java]: This doesn't seem right, why are these necessary?

File: biojava-genome/src/test/java/org/biojava/nbio/genome/io/fastq/AbstractFastqWriterTest.java [hunk 3]
```diff
@@ -28,6 +28,9 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import org.biojava.nbio.genome.io.fastq.Fastq;
```
Comment (heuermh) [inline biojava-genome/src/test/java/org/biojava/nbio/genome/io/fastq/AbstractFastqWriterTest.java]: ... similar here

File: biojava-genome/src/test/java/org/biojava/nbio/genome/io/fastq/ConvertTest.java [hunk 4]
```diff
@@ -18,25 +18,35 @@
...
-package org.biojava.nbio.sequencing.io.fastq;
+package org.biojava.nbio.genome.io.fastq;
 
 import java.io.File;
 import java.io.FileWriter;
 import java.util.List;
 import java.util.Map;
 
-import junit.framework.TestCase;
+import org.biojava.nbio.genome.io.fastq.Fastq;
```
Comment (heuermh) [inline biojava-genome/src/test/java/org/biojava/nbio/genome/io/fastq/ConvertTest.java]: ... did these all come in via an IDE?

File: biojava-genome/src/main/java/org/biojava/nbio/genome/io/fastq/AbstractFastqReader.java (hunk 1 above)
Comment (josemduarte) [reply biojava-genome/src/main/java/org/biojava/nbio/genome/io/fastq/AbstractFastqReader.java]: Good point, I'm not sure how to do `@since` tags for these cases where something existed but was moved, rather than appeared new. Ideas anyone?

File: biojava-genome/src/test/java/org/biojava/nbio/genome/io/fastq/ConvertTest.java (hunk 4 above)
Comment (josemduarte) [reply biojava-genome/src/test/java/org/biojava/nbio/genome/io/fastq/ConvertTest.java]: Somehow eclipse considered them necessary... They are indeed redundant. I'll push a commit.

**Comment by heuermh** [comment]
Ping me if this merges before #740; I'll rebase that pull request.

**Comment by heuermh** [comment]
Note #740 went first, so the Guava dependency version in the genome module pom should also be bumped to `<version>24.0-jre</version>`, I didn't catch it there.

**Comment by josemduarte** [comment]
Thanks @heuermh, I'll push a commit to solve the conflicts.

File: biojava-genome/src/main/java/org/biojava/nbio/genome/io/fastq/AbstractFastqReader.java (hunk 1 above)
Comment (lafita) [reply biojava-genome/src/main/java/org/biojava/nbio/genome/io/fastq/AbstractFastqReader.java]: I would not use the `@since` tag, because it means that the class is new. Maybe just add a line like:

`moved from org.biojava.nbio.sequencing in 5.0.0`

File: biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java [hunk 5]
```diff
@@ -610,7 +593,8 @@ public static int getCDSLengthForward(List<Integer> exonStarts, List<Integer> ex
         if ( debug)
             s = new StringBuffer();
 
-        int lengthExons = 0;
+        @SuppressWarnings("unused")
+		int lengthExons = 0;
```
Comment (lafita) [inline biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java]: Why is this variable not used? Just wondering if it would be better to remove it to eliminate confusion.

**Comment by josemduarte** [comment]
Thanks for comments @lafita , I've pushed a commit addressing them.

**Comment by josemduarte** [comment]
Tested, I'll merge

**Comment by heuermh** [comment]
Thanks!
